plugins/filter_ecs/ecs.c

```diff
@@ -341,24 +347,73 @@ static int flb_ecs_metadata_buffer_init(struct flb_filter_ecs *ctx,
         return -1;
     }
 
-    meta->unpacked = result;
-    meta->obj = root;
+    /* create list of metadata key value pairs */
+    mk_list_init(&meta->metadata_keypairs);
+    meta->keypairs_len = 0;
+
     meta->last_used_time = time(NULL);
-    meta->free_packer = FLB_TRUE;
+
+    mk_list_foreach_safe(head, tmp, &ctx->metadata_keys) {
+        metadata_key = mk_list_entry(head, struct flb_ecs_metadata_key, _head);
+        val = flb_ra_translate(metadata_key->ra, NULL, 0,
+                               root, NULL);
+        if (!val) {
+            flb_plg_info(ctx->ins, "Translation failed for %s : %s for container ID: %s",
+                         metadata_key->key, metadata_key->template, meta->id);
+            /* keep trying other keys*/
+            continue;
+        }
+
+        if (flb_sds_len(val) == 0) {
+            flb_plg_info(ctx->ins, "Translation failed for %s : %s for container ID: %s",
+                         metadata_key->key, metadata_key->template, meta->id);
+            flb_sds_destroy(val);
+            /* keep trying other keys*/
+            continue;
+        }
```

Comment:
I suppose value will never be null or length 0

Reply:
(this was to address the other comment asking if we need to check some val==null condition as we do in other places)

Reply:
uhhh its null if the record accessor translation fails

Reply:
I'm not following you here.